superset/assets/spec/javascripts/visualizations/table_spec.jsx

```diff
@@ -1,100 +1,98 @@
 import $ from 'jquery';
 import '../../helpers/shim';
-import tableVis from '../../../src/visualizations/Table/adaptor';
+import Table from '../../../src/visualizations/Table/Table';
+import transformProps from '../../../src/visualizations/Table/transformProps';
 
 describe('table viz', () => {
   const div = '<div id="slice-container"><div class="dataTables_wrapper"></div></div>';
-  const baseSlice = {
-    selector: '#slice-container',
+  const BASE_CHART_PROPS = {
+    height: 100,
+    datasource: {
+      verboseMap: {},
+    },
+    filters: {},
     formData: {
       metrics: ['count'],
-      timeseries_limit_metric: null,
+      timeseriesLimitMetric: null,
     },
-    datasource: {
-      verbose_map: {},
+    onAddFilter() {},
+    payload: {
+      data: {
+        records: [
+          { gender: 'boy', count: 39245 },
+          { gender: 'girl', count: 36446 },
+        ],
+        columns: ['gender', 'count'],
+      },
     },
-    getFilters: () => ({}),
-    removeFilter() {},
-    addFilter() {},
-    width: () => 0,
-    height: () => 0,
   };
-  const basePayload = {
+
+  const PAYLOAD2 = {
     data: {
       records: [
-        { gender: 'boy', count: 39245 },
-        { gender: 'girl', count: 36446 },
+        { gender: 'boy', count: 39245, 'SUM(sum_boys)': 48133355 },
+        { gender: 'girl', count: 36446, 'SUM(sum_boys)': 0 },
       ],
-      columns: ['gender', 'count'],
+      columns: ['gender', 'count', 'SUM(sum_boys)'],
     },
   };
 
-  it('renders into a container', () => {
+  let container;
+  let $container;
+
+  beforeEach(() => {
     $('body').html(div);
-    const container = $(baseSlice.selector);
-    expect(container).toHaveLength(1);
+    container = document.getElementById('slice-container');
+    $container = $(container);
   });
 
-  it('renders header and body datatables in container', () => {
-    $('body').html(div);
-    const container = $(baseSlice.selector);
+  it('renders into a container', () => {
+    expect($container).toHaveLength(1);
```

Comment:
is this testing the right thing? I see that you just translated the logic, but I think by definition if you select something by `id` it will have a length `1`? 

should the test instead assert that the container child nodes array have length `1`?

Reply:
Let me check. Wasn't paying attention to logic. Now agree it looks weird